templates/media/media--image.html.twig

```diff
@@ -0,0 +1,50 @@
+{#
+/**
+ * @file
+ * Theme override to display an image media item with WA-compliant description.
+ * ~ When field_aria_description has value, this template:
+ * ~~ Adds aria-describedby="description-[MID]" to field_media_image <img> tag.
+ * ~~ Renders field_aria_description inside div#description-[MID].sr-only.
+ * See cwd_base/media/media.html.twig for available variables and other info.
+ *
+ * @see template_preprocess_media()
+ *
+ * @ingroup themeable
+ */
+#}
+{{ title_suffix.contextual_links }}
+{% if content %}
+
+  {% if media.field_aria_description.0.value -%}
+
+    {####### variable setup #######}
+    {% set media_id = media.mid.0.value %}
+    {% set technical_image = {} %}
+    {% if media.field_media_image.0.value -%}
+      {% set describedby = 'description-' ~ media_id %}
+      {# Get the image style for whatever view mode was selected when embedding this image. #}
+      {% set style_name = content.field_media_image[0]['#image_style'] %}
+      {# Create the image. #}
+      {% set technical_image = {
+        '#theme':      'image_style',
+        '#style_name': style_name,
+        '#uri':        media.field_media_image.entity.uri.value,
+        '#alt':        media.field_media_image[0].alt,
+        '#attributes': {'aria-describedby':[describedby]},
+      } %}
+    {% endif %}
+
+    {####### output #######}
+    {{ technical_image }}
+    <div id="description-{{ media_id }}" class="sr-only">
+      {{ media.field_aria_description|view }}
```

Comment:
The `view` filter (from twig_tweak) defaults to using the "default" view mode (from the image media bundle display settings).
By using `media.field_aria_description|view` instead of `content.field_aria_description`, the field will be rendered no matter which display option the user picks when embedding an image in a WYSIWYG field (without relying on field_aria_description being enabled on each view mode) -- along with images attached via entity reference fields.

As usual, when different behavior is desired, this template can be overridden by more specific templates.